S/Sundials/build_tarballs.jl

```diff
@@ -1,31 +1,29 @@
 using BinaryBuilder
 
 name = "Sundials"
-version = v"3.1.1"
+version = v"5.1.0"
 
 # Collection of sources required to build SundialsBuilder
 sources = [
     "https://github.com/LLNL/sundials/archive/v$(version).tar.gz" =>
     "d03fca24d7adbfe52e05bd4e61340251e1cf56927540cc77a1ca817716091166",
-    "./bundled",
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/sundials-*/
-patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
+# patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
 
 CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}")
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
 CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
-CMAKE_FLAGS+=(-DBLAS_ENABLE=OFF -DLAPACK_ENABLE=OFF)
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
 
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
```

Comment:
We still should need this patch to mangle the BLAS and LAPACK names correctly to use our `64_` prefixes.

Reply:
That's now done in a different file `config/SundialsFortran`. I have a patch file but didn't add it yet.

Reply:
Could you add it? That way we can clone this PR and see how it goes.

Reply:
In the meanwhile let's see what we need for the 32-bit builders to go through.

Reply:
It looks like most of the platforms that don't have 64-bit integers OpenBLAS are going well.

Reply:
we need to check that BLAS and LAPACK are detected as operational by Sundials.

Reply:
@giordano https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1117&view=logs&j=8e1f3fbd-1eec-58ce-53a3-f5a5154dbbe1&t=753643ea-faac-5ab7-7043-7b47880b417b&l=1717

Reply:
In #419 I'm `grep`ping `config.log` to make sure that some libraries we need are correctly identified.  You can do something similar, if you find something usable in the same in the gazillions of files that CMake generates.

Reply:
> We still should need this patch to mangle the BLAS and LAPACK names correctly to use our `64_` prefixes.

@ViralBShah added in da9aa31 but can't tell why is causing the pipeline to fail